src/netconf_acm.h

```diff
@@ -0,0 +1,172 @@
+
+/**
+ * @brief Main NACM container structure.
+ */
+struct ncac {
+    char enabled;                   /**< Whether NACM is enabled. */
+    char default_read_deny;         /**< Whether default NACM read action is "deny" (otherwise "permit"). */
+    char default_write_deny;        /**< Whether default NACM write action is "deny" (otherwise "permit"). */
+    char default_exec_deny;         /**< Whether default NACM exec action is "deny" (otherwise "permit"). */
+    char enable_external_groups;    /**< Whether external (system) groups are taken into consideration for NACM. */
+
+    uint32_t denied_operations;     /**< Counter of denied operations (RPC or action). */
+    uint32_t denied_data_writes;    /**< Counter of denied data writes. */
+    uint32_t denied_notifications;  /**< Counter of denied notifications. */
+
+    /**
+     * @brief NACM group.
+     */
+    struct ncac_group {
+        char *name;                 /**< Group name. */
+        char **users;               /**< Array of users belonging to this group. */
+        uint32_t user_count;        /**< Number of users. */
+    } *groups;                      /**< Sorted array of existing groups. */
+    uint32_t group_count;           /**< Number of groups. */
+
+    /**
+     * @brief NACM rule list.
+     */
+    struct ncac_rule_list {
+        char *name;                 /**< Rule list name. */
+        char **groups;              /**< Sorted all groups associated with this rule list. */
+        uint32_t group_count;       /**< Number of groups. */
+
+        /**
+         * @brief NACM rule.
+         */
+        struct ncac_rule {
+            char *name;             /**< Rule name. */
+            char *module_name;      /**< Rule module name. */
+            char *target;           /**< Rule target. */
+            NCAC_TARGET_TYPE target_type;   /**< Rule target type. */
+            uint8_t operations;     /**< Rule operations associated with it. */
+            char action_deny;       /**< Whether the rule action is "deny" (otherwise "permit"). */
+            char *comment;          /**< Rule comment. */
+            struct ncac_rule *next; /**< Pointer to the next rule. */
+        } *rules;                   /**< List of rules in the rule list. */
+
+        struct ncac_rule_list *next;    /**< Pointer to the next rule list. */
+    } *rule_lists;                  /**< List of all the rule lists. */
+
+    pthread_mutex_t lock;           /**< Lock for accessing all the NACM members. */
+};
+
+enum ncac_access {
+    NCAC_ACCESS_DENY = 1,           /**< access to the node is denied */
+    NCAC_ACCESS_PARTIAL_DENY = 2,   /**< access to the node is denied but it is a prefix of a matching rule */
+    NCAC_ACCESS_PARTIAL_PERMIT = 3, /**< access to the node is permitted but any children must still be checked */
+    NCAC_ACCESS_PERMIT = 4          /**< access to the node is permitted with any children */
+};
+
+#define NCAC_ACCESS_IS_NODE_PERMIT(x) ((x) > 2)
```

Comment:
It will now be part of *sysrepo* so it needs `sr_` prefix just like everything else, all the function in the internal API at least.

Reply:
done